src/hooks/custom-hooks/useOAuth.ts

```diff
@@ -19,34 +26,54 @@ const useOAuth = (): UseOAuthReturn => {
         featureFlag: 'hydra_be',
     }) as unknown as [TOAuth2EnabledAppList, boolean];
 
-    const oAuthGrowthbookConfig = {
-        OAuth2EnabledApps,
-        OAuth2EnabledAppsInitialised,
-    };
+    const isOAuth2Enabled = useIsOAuth2Enabled(OAuth2EnabledApps, OAuth2EnabledAppsInitialised);
 
     const { logout } = useAuthData();
     const { error, isAuthorized, isAuthorizing } = useAuthData();
     const isEndpointPage = getCurrentRoute() === 'endpoint';
+    const isCallbackPage = getCurrentRoute() === 'callback';
     const oauthUrl = getOauthUrl();
 
     const WSLogoutAndRedirect = async () => {
         await logout();
         removeCookies('affiliate_token', 'affiliate_tracking', 'utm_data', 'onfido_token', 'gclid');
-        window.open(oauthUrl, '_self');
+        if (isOAuth2Enabled) {
+            await requestOidcAuthentication({
+                redirectCallbackUri: `${window.location.origin}/callback`,
+            });
+        } else {
+            window.open(oauthUrl, '_self');
+        }
     };
-    const { OAuth2Logout: oAuthLogout } = useOAuth2(oAuthGrowthbookConfig, WSLogoutAndRedirect);
+    const handleLogout = async () => {
+        await OAuth2Logout(WSLogoutAndRedirect);
+    };
+
+    const redirectToAuth = async () => {
+        if (isOAuth2Enabled) {
+            await requestOidcAuthentication({
+                redirectCallbackUri: `${window.location.origin}/callback`,
+            });
+        } else {
+            window.open(oauthUrl, '_self');
+        }
+    };
+
+    const hasAuthToken = localStorage.getItem('authToken');
+    const loggedState = Cookies.get('logged_state');
 
-    const onRenderAuthCheck = useCallback(() => {
-        if (!isEndpointPage) {
-            if (error?.code === 'InvalidToken') {
-                oAuthLogout();
+    const onRenderAuthCheck = useCallback(async () => {
+        if (!isEndpointPage && !isCallbackPage) {
+            if ((hasAuthToken && loggedState === 'false' && isOAuth2Enabled) || error?.code === 'InvalidToken') {
+                await handleLogout();
             } else if (!isAuthorized && !isAuthorizing) {
-                window.open(oauthUrl, '_self');
+                await redirectToAuth();
             }
         }
-    }, [isEndpointPage, error?.code, isAuthorized, isAuthorizing, oAuthLogout, oauthUrl]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
nit: I don't think we need to `useCallback` this function anymore, it has lots of dependencies already 😅